app/src/main/java/org/coepi/android/cen/CenRepo.kt

```diff
@@ -13,17 +16,40 @@ import javax.crypto.KeyGenerator
 import javax.crypto.SecretKey
 import javax.crypto.spec.SecretKeySpec
 
+interface CenRepo {
+     // Periodically generated CEN, to be made available via BLE
+    val generatedCen: Observable<Cen>
+
+    // Infection reports fetched periodically from the API
+    val reports: Observable<List<CenReport>>
+
+     // Send symptoms report
+    fun sendReport(report: CenReport): Completable
+
+    // Store CEN from other device
+    fun storeCen(cen: Cen)
+}
```

Comment:
Thanks for taking the initiative, I didn't realize you could make a library have `Observable` and `Cen`, `CenReports`.   I would 
1. verify that the Rust library can conform to this interface
2. make sure that the iOS interface is basically the same and also has the same idea
to ratify the idea.  If it can't, you won't achieve the vision you are looking for -- I think you can have a stub that does it on the other side.

Reply:
1. That's what the (Kotlin) "glue" (the implementation of this interface) solves. It bridges between callbacks and rx.
2. Swift is totally doable. It's a question of how much time we have for the needed refactorings before 0.1. But maybe someone is idle, or I can work on it.